add recipe for soccer

md-arif-shaikh opened the pull request:
### Brief summary of what the package does

This package brings soccer fixtures, results in Emacs.

### Direct link to the package repository

https://github.com/md-arif-shaikh/soccer

### Your association with the package

Maintainer. Author.

### Relevant communications with the upstream package maintainer

*None needed**

### Checklist

<!-- Please confirm with `x`: -->

- [ x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [ x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [ x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [ x] My elisp byte-compiles cleanly
- [ x] `M-x checkdoc` is happy with my docstrings
- [ x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [ ] I have confirmed some of these without doing them

<!-- After submitting, please fix any problems the CI reports. -->

riscy commented:
Quick first pass --

### soccer-leagues.el ###

byte-compile (using Emacs 27.1):
- No issues!

checkdoc (using version 0.6.2):
- No issues!

package-lint (using version 20210528.2348):
```
2 issues found:
7:0: error: Package-Requires outside the main file have no effect.
8:3: warning: You should include standard keywords: see the variable `finder-known-keywords'.
```
The `Package-Requires` error can come as a surprise.  It requires you to do some setup around the variable `package-lint-main-file` but it's probably not a blocker.  I believe setting keywords is unnecessary in non-main files.

### soccer-time.el ###

byte-compile (using Emacs 27.1):
- No issues!

checkdoc (using version 0.6.2):
```
Warning (emacs): 
soccer-time.el:8: White space found at end of line
```

package-lint (using version 20210528.2348):
```
2 issues found:
7:0: error: Package-Requires outside the main file have no effect.
8:3: warning: You should include standard keywords: see the variable `finder-known-keywords'.
```

Other [lints](https://github.com/riscy/melpazoid):
- soccer-time.el#L30: Customizable variables shouldn't be private
- soccer-time.el#L35: Customizable variables shouldn't be private

### soccer.el ###

byte-compile (using Emacs 27.1):
- No issues!

checkdoc (using version 0.6.2):
- No issues!

package-lint (using version 20210528.2348):
```
2 issues found:
1:0: warning: "Version:" or "Package-Version:" header is missing. MELPA will handle this, but other archives will not.
8:3: warning: You should include standard keywords: see the variable `finder-known-keywords'.
```
Prefer `Version` over `Package-Version`, and this is the file where you'd want to address the `finder-known-keywords` warning.

md-arif-shaikh commented:
Hi, thanks for the checks. `Customizable variables shouldn't be private` what does it exactly mean? Does it mean that it should be inside the main file? Could you kindly elaborate?

riscy commented:
> I have ignored error: Package-Requires ...

Yeah, I think that's fine.  Just be aware that MELPA will ignore the ones that aren't in soccer.el.  If you want to fix that, see the documentation for `package-lint-main-file`.  You could use .dir-locals or add the following somewhere in each of your secondary files (just make sure that it isn't inside your `;;; Commentary` block):

```elisp
;; Local variables:
;; package-lint-main-file: "my-main-file.el"
;; end:
```
but that's just fyi.

> pending Customizable variables shouldn't be private.

Private variables are ones with two hyphens in them, e.g. `soccer--private-variable`.  Those two hyphens signal the variable isn't part of your public interface.  But making them customizable puts them on a list that is immediately visible and togglable by users, which sort of defeats the point.

md-arif-shaikh commented:
Thanks, @riscy for the clarifications. I have now added the variable `package-lint-main-file` in the secondary files and also removed the double hyphens from those customizable variables. All issues should now be fixed.

riscy commented:
Great thanks!  Another quick pass --

- I think you added a dependency on `dash` that isn't in your `Package-Requires`.
- 1:0: error: package.el cannot parse this buffer: Search failed: ";;; soccer-leauges.el ends here"

md-arif-shaikh commented:
I don't understand why `package.el` fails to parse the buffer. Do you have any idea what could cause this?

md-arif-shaikh commented:
Hi @riscy, I have now fixed all the issues you mentioned.

riscy commented:
Thanks, that puts the checklist behind us.

Taking a closer look for what the linters missed:
- All of your interactive functions shouldn't be private -- e.g. `soccer--fixtures-full-in-org` -- by virtue of being interactive these are part of your public interface (the `--` in the function name indicates 'private')
- Similarly, faces like `soccer-face--win` shouldn't be private (users will want to customize them, themes might want to define them
- soccer.el#L240: Use `temporary-file-directory` instead of /tmp in code
- soccer.el: Minor, but for readability I notice several large expressions of the form `(string-join (list (format x) (format y) ...))` that could be replaced with a `concat` or even a simple `format`.  For example:

```elisp
(string-join (list "Rank" "Team" "Played" "Won" "Lost" "Drawn" (format "%3s" "GF") (format "%3s" "GA") (format "%3s" "GD") "Points" (format "%5s" "Form")) " ")
```
could be:
```
(format "Rank Team Played Won Lost Drawn %3s %3s %3s Points %5s"
        "GF" "GA" "GD" "Form")
```
(but I'd be tempted to remove the spacing patterns entirely).

md-arif-shaikh commented:
Thanks for these comments, I have taken care of all these points. Btw, I did not get rid of the spacing patterns completely since I found that the table alignment works better with these and I don't know how to get that using something else.
